meshroom/core/node.py

```diff
@@ -1463,5 +1464,7 @@ def nodeFactory(nodeDict, name=None):
         if not internalFolder and nodeDesc:
             logging.warning("No serialized output data: performing automatic upgrade on '{}'".format(name))
             node = node.upgrade()
+        if template:  # if the node comes from a template file and there is a conflict, it should be upgraded anyway
```

Comment:
Maybe `elif` to ensure that we are not calling the upgrade twice in any case?

Reply:
Done